plugins/catalog-backend/src/service/DefaultEntitiesCatalog.ts

```diff
@@ -204,27 +200,6 @@ export class DefaultEntitiesCatalog implements EntitiesCatalog {
       entities = entities.map(e => request.fields!(e));
     }
 
-    // TODO(freben): This is added as a compatibility guarantee, until we can be
-    // sure that all adopters have re-stitched their entities so that the new
-    // targetRef field is present on them, and that they have stopped consuming
-    // the now-removed old field
-    for (const entity of entities) {
-      if (entity.relations) {
-        for (const relation of entity.relations) {
-          if (!relation.targetRef && relation.target) {
-            // This is the case where an old-form entity, not yet stitched with
-            // the updated code, was in the database
-            relation.targetRef = stringifyEntityRef(relation.target);
-          } else if (!relation.target && relation.targetRef) {
-            // This is the case where a new-form entity, stitched with the
-            // updated code, was in the database but we still want to produce
-            // the old data shape as well for compatibility reasons
-            relation.target = parseEntityRef(relation.targetRef);
-          }
-        }
-      }
-    }
```

Comment:
Actually, maybe we should keep this piece of code in here, for one or two more release cycles. I want to be _dead sure_ that people have reprocessed their entities and aren't in a half-baked state where some of the entities make their consumers explode and others don't. Especially those that maybe don't update every single week.

Reply:
`relation.target` is no longer available with this change so the code can't remain in its current form.

Reply:
made relations `any` to keep the compatibility layer